lib/utils.js

```diff
@@ -33,7 +33,12 @@ exports.inherits = util.inherits;
  * @return {string}
  */
 exports.escape = function (html) {
-  return he.encode(String(html), {useNamedReferences: false});
+  return he
+    .encode(String(html), {useNamedReferences: false})
+    .replace(/&#x([0-9A-F]);/g, (match) => {
+      const val = Number.parseInt(match); 
+      return val === 0x9 || val === 0xA || val === 0xD || (val >= 0xE000 && val <= 0xFFFD) || (val >= 0x10000 && val <= 0x10FFFF) ? `&#x${match};` : '';
+    });
```

Comment:
Suggestion:

Turns out that `he` already has this in itself, just doesn't expose it other than through `strict: true`, which throws when it encounters it.

Maybe we could use that regexp and try to land a PR that either exposes it or adds a `skipInvalid: true` option or such?

```suggestion
  return  he.encode(String(html).replace(regexInvalidRawCodePoint, ''), {useNamedReferences: false});
```

Requires this to be added as well:

```js
// From https://github.com/mathiasbynens/he/blob/36afe179392226cf1b6ccdb16ebbb7a5a844d93a/he.js#L53
var regexInvalidRawCodePoint = /[\0-\x08\x0B\x0E-\x1F\x7F-\x9F\uFDD0-\uFDEF\uFFFE\uFFFF]|[\uD83F\uD87F\uD8BF\uD8FF\uD93F\uD97F\uD9BF\uD9FF\uDA3F\uDA7F\uDABF\uDAFF\uDB3F\uDB7F\uDBBF\uDBFF][\uDFFE\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]/g;
```